% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={homework\_01},
  pdfauthor={Esteban Jorquera},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{homework\_01}
\author{Esteban Jorquera}
\date{2022-02-13}

\begin{document}
\maketitle

\hypertarget{practical-bam-file}{%
\subsection{Practical: BAM file}\label{practical-bam-file}}

\begin{verbatim}
# opens a qlogin session 
qlogin

# copies the NA20538.bam file from its parent directory to a local work directory
cp /mnt/Timina/bioinfoII/data/format_qc/NA20538.bam /mnt/Citosina/amedina/ejorquera/BioInfoII/

# checks module availability   
module avail

# loads a version (1.9) of the required program samtools, available from the module list
module load samtools/1.9

# displays samtools view options
samtools view -man

# loads the bam file and print only the header
samtools view -H NA20538.bam | less -S
\end{verbatim}

\hypertarget{what-does-rg-stand-for}{%
\section{What does RG stand for?}\label{what-does-rg-stand-for}}

RG stands for read group

\hypertarget{what-is-the-lane-id-lane-is-the-basic-independent-run-of-a-high--throughput-sequencing-machine.-for-illumina-machines-this-is-the-physical-sequencing-lane.-reads-from-one-lane-are-identified-by-the-same-read-group-id-and-the-information-about-lanes-can-be-found-in-the-header-in-lines-starting-with-rg.}{%
\section{What is the lane ID? (``Lane'' is the basic independent run of
a high- throughput sequencing machine. For Illumina machines, this is
the physical sequencing lane. Reads from one lane are identified by the
same read group ID and the information about lanes can be found in the
header in lines starting with
``@RG''.)}\label{what-is-the-lane-id-lane-is-the-basic-independent-run-of-a-high--throughput-sequencing-machine.-for-illumina-machines-this-is-the-physical-sequencing-lane.-reads-from-one-lane-are-identified-by-the-same-read-group-id-and-the-information-about-lanes-can-be-found-in-the-header-in-lines-starting-with-rg.}}

The lane ID is referenced by the ``ID'' tag

\hypertarget{what-is-the-sequencing-platform}{%
\section{What is the sequencing
platform?}\label{what-is-the-sequencing-platform}}

The sequencing platform is referenced by the ``PL'' tag, according to
this, the used platform was Illumina

\hypertarget{what-version-of-the-human-assembly-was-used-to-perform-the-alignments}{%
\section{What version of the human assembly was used to perform the
alignments?}\label{what-version-of-the-human-assembly-was-used-to-perform-the-alignments}}

Found at the @SQ lines (Reference sequence dictionary), referenced in
the ``AS'' tag (NCBI37) and in the ``UR'' tag
(\url{ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/human_g1k_v37.fasta.gz}),
according to them, the reference genome assembly used corresponds to
GRCh37 (Genome Reference Consortium Human Build 37), specifically the
one downloaded from the 1000 Genomes Project

\hypertarget{what-programs-were-used-to-create-this-bam-file}{%
\section{What programs were used to create this BAM
file?}\label{what-programs-were-used-to-create-this-bam-file}}

Found at each @PG line, referenced in the ``ID'' tags (program name) and
``VN'' tags (version number), the used programs corresponded to: -Genome
Analysis Toolkit (GATK) - IndelRealigner version 1.0.4487 -Genome
Analysis Toolkit (GATK) - TableRecalibration version 1.0.4487
-Burrows-Wheeler Aligner (bwa) version 0.5.5

\hypertarget{what-version-of-bwa-was-used-to-align-the-reads}{%
\section{What version of bwa was used to align the
reads?}\label{what-version-of-bwa-was-used-to-align-the-reads}}

As mentioned before, the used version of bwa corresponds to version
0.5.5, as mentioned in its ``VN'' tag

\hypertarget{what-is-the-name-of-the-first-read}{%
\section{What is the name of the first
read?}\label{what-is-the-name-of-the-first-read}}

\begin{verbatim}
# loads the bam file and also prints the header
samtools view -h NA20538.bam | less -S
\end{verbatim}

As shown in the first line that does not correspond to the header, the
first read is named ERR003814.1408899

\hypertarget{what-position-does-the-alignment-of-the-read-start-at}{%
\section{What position does the alignment of the read start
at?}\label{what-position-does-the-alignment-of-the-read-start-at}}

The read ERR003814.1408899 starts at position 19999970 as it is shown in
the fourth column (POS)

\hypertarget{what-is-the-mapping-quality-of-the-first-read}{%
\section{What is the mapping quality of the first
read?}\label{what-is-the-mapping-quality-of-the-first-read}}

The mapping quality of the read ERR003814.1408899 corresponds to 23, as
it is shown in the fifth column (MAPQ)

\#\#Practical bcf

\hypertarget{what-is-a-bcf}{%
\section{What is a bcf?}\label{what-is-a-bcf}}

A bcf file is the binary variant of a vcf file (Variant Call Format),
which in itself is a text file format for the storage of gene sequence
variations, therefore a bcf is a Binary variant calling format

\hypertarget{can-you-convert-bcf-to-vcf-using-bcftools-how}{%
\section{Can you convert bcf to vcf using bcftools?
How?}\label{can-you-convert-bcf-to-vcf-using-bcftools-how}}

Yes, by executing the following code

\begin{verbatim}
# copies the 1kg.bcf file, and its index file from their parent directory to a local work directory
cp /mnt/Timina/bioinfoII/data/format_qc/1kg.bcf /mnt/Citosina/amedina/ejorquera/BioInfoII/
cp /mnt/Timina/bioinfoII/data/format_qc/1kg.bcf.csi /mnt/Citosina/amedina/ejorquera/BioInfoII/

# checks module availability 
module avail

# loads a version (1.10.2) of the required program bcftools, available from the module list
module load bcftools/1.10.2

# loads the bcf file and saves it as a vcf file
bcftools view /mnt/Citosina/amedina/ejorquera/BioInfoII/1kg.bcf > 1kg.vcf
\end{verbatim}

Using the view command will load the binary file by bcftools, then
saving it as a vcf file will make it human readable

\hypertarget{how-many-samples-are-in-the-bcf-hint-use-the--l-option.}{%
\section{How many samples are in the BCF? Hint: use the -l
option.}\label{how-many-samples-are-in-the-bcf-hint-use-the--l-option.}}

\begin{verbatim}
# asks the bcf file for the sample list and counts the lines present
bcftools query -l /mnt/Citosina/amedina/ejorquera/BioInfoII/1kg.bcf | wc -l
\end{verbatim}

query extracts fields from either the vcf or bcf file. The -l option
makes query output the list of samples in the file, piping the previous
with ``wc -l'' makes it count the number of lines in the output.

So therefore, there are 50 samples in the file

\hypertarget{what-is-the-genotype-of-the-sample-hg00107-at-the-position-2024019472-hint-use-the-combination-of--r--s-and--f-tgt-options.}{%
\section{\texorpdfstring{What is the genotype of the sample HG00107 at
the position 20:24019472? Hint: use the combination of -r, -s, and -f
`{[} \%TGT{]}\n'
options.}{What is the genotype of the sample HG00107 at the position 20:24019472? Hint: use the combination of -r, -s, and -f `{[} \%TGT{]}' options.}}\label{what-is-the-genotype-of-the-sample-hg00107-at-the-position-2024019472-hint-use-the-combination-of--r--s-and--f-tgt-options.}}

\begin{verbatim}
# shows the data contained in the bcf file for the HG00107 sample at the chromosome 20 position 24019472
bcftools view -r 20:24019472 -s HG00107 -f '[ %TGT]\n' /mnt/Citosina/amedina/ejorquera/BioInfoII/1kg.bcf
\end{verbatim}

At the position 20:24019472, the sample HG00107 has a heterozygous
phenotype as indicated by the ``0/1'' data of the ``GT'' tag, we can't
be sure in which chromosome is located the reference (A) or alternative
(T) alleles due to the use of the ``/'' symbol instead of ``\textbar{}''

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{table }\OtherTok{\textless{}{-}} \FunctionTok{matrix}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{\textquotesingle{}20\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}24019472\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}.\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}A\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}T\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}999\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}.\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}AN=2;AC=1;AC\_Het=16;AC\_Hom=2;AC\_Hemi=0\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}GT:PL:DP\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}0/1:235;0;148:16\textquotesingle{}}\NormalTok{), }\AttributeTok{ncol=}\DecValTok{10}\NormalTok{, }\AttributeTok{byrow=}\ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{colnames}\NormalTok{(table) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{\textquotesingle{}\#CHROM\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}POS\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}ID\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}REF\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}ALT\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}QUAL\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}FILTER\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}INFO\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}FORMAT\textquotesingle{}}\NormalTok{,}\StringTok{\textquotesingle{}HG00107\textquotesingle{}}\NormalTok{)}
\NormalTok{table\_k }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(table)}
\FunctionTok{kable}\NormalTok{(table\_k,}\AttributeTok{align =} \StringTok{"c"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{longtable}[]{@{}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.07}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.09}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.04}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.04}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.04}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.05}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.07}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.35}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.09}}
  >{\centering\arraybackslash}p{(\columnwidth - 18\tabcolsep) * \real{0.16}}@{}}
\toprule
\#CHROM & POS & ID & REF & ALT & QUAL & FILTER & INFO & FORMAT &
HG00107 \\
\midrule
\endhead
20 & 24019472 & . & A & T & 999 & . &
AN=2;AC=1;AC\_Het=16;AC\_Hom=2;AC\_Hemi=0 & GT:PL:DP &
0/1:235;0;148:16 \\
\bottomrule
\end{longtable}

\hypertarget{how-many-positions-there-are-with-more-than-10-alternate-alleles-see-the-infoac-tag.-hint-use-the--i-filtering-option.}{%
\section{How many positions there are with more than 10 alternate
alleles? (See the INFO/AC tag.) Hint: use the -i filtering
option.}\label{how-many-positions-there-are-with-more-than-10-alternate-alleles-see-the-infoac-tag.-hint-use-the--i-filtering-option.}}

\begin{verbatim}
# asks the bcf file for any position with more than 10 alternate alleles and counts the lines present
bcftools query -f '%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n' -i 'AC>10' /mnt/Citosina/amedina/ejorquera/BioInfoII/1kg.bcf | wc -l
\end{verbatim}

query shows the data contained in the bcf file in the -f `\#', format -f
'\%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n' set the format of the query output to chromosome, position, reference allele, alternative allele, and the sample's genotype, -i 'AC>10' sets a filter to only display the lines with an "AC" value bigger than 10, piping the previous with "wc -l" makes it count the number of lines in the output.

So therefore, there are 4778 positions with more than 10 alternate
alleles

\hypertarget{list-all-positions-where-hg00107-has-a-non-reference-genotype-and-the-read-depth-is-bigger-than-10.}{%
\section{List all positions where HG00107 has a non-reference genotype
and the read depth is bigger than
10.}\label{list-all-positions-where-hg00107-has-a-non-reference-genotype-and-the-read-depth-is-bigger-than-10.}}

\begin{verbatim}
bcftools query -f '%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n' -s HG00107 -i 'DP>10 && GT="ref"' /mnt/Citosina/amedina/ejorquera/BioInfoII/1kg.bcf | wc -l
\end{verbatim}

query shows the data contained in the bcf file in the -f `\#' format, -f
'\%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n' set the format of the query output to chromosome, position, reference allele, alternative allele, and the sample's genotype, -s HG00107 set the samples to be evaluated to only HG00107, -i 'DP>10 && GT="alt"' sets a filter to only display the lines with a depth ("DP") bigger than 10 and where the genotype ("GT") is not equal to the reference, piping the previous with "wc -l" makes it count the number of lines in the output.

So therefore, there are 451 positions where the sample HG00107 has a
non-reference genotype and the read depth is bigger than 10.

\#\#Practical stats

\hypertarget{using-samtools-stats-answer}{%
\section{Using samtools stats
answer}\label{using-samtools-stats-answer}}

\begin{verbatim}
# displays samtools stats options
samtools stats -man
\end{verbatim}

\hypertarget{what-is-the-total-number-of-reads}{%
\section{What is the total number of
reads?}\label{what-is-the-total-number-of-reads}}

\begin{verbatim}
# displays samtools stats options
samtools stats -man

# shows stats for the NA20538.bam file and uses grep and cut to only show the summary numbers
samtools stats NA20538.bam | grep ^SN | cut -f 2-
\end{verbatim}

\hypertarget{what-proportion-of-the-reads-were-mapped}{%
\section{What proportion of the reads were
mapped?}\label{what-proportion-of-the-reads-were-mapped}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# total reads results of previous code}
\NormalTok{reads\_raw }\OtherTok{\textless{}{-}} \DecValTok{347367}
\CommentTok{\# total mapped reads results of previous code}
\NormalTok{reads\_mapped }\OtherTok{\textless{}{-}} \DecValTok{323966}
\CommentTok{\# percent of the total reads that were mapped}
\NormalTok{reads\_percent }\OtherTok{=}\NormalTok{ reads\_mapped}\SpecialCharTok{*}\DecValTok{100}\SpecialCharTok{/}\NormalTok{reads\_raw}
\end{Highlighting}
\end{Shaded}

The portion of reads that were mapped corresponds to 93.3\%

\hypertarget{how-many-reads-were-mapped-to-a-different-chromosome}{%
\section{How many reads were mapped to a different
chromosome?}\label{how-many-reads-were-mapped-to-a-different-chromosome}}

According to the summary data of the NA20538.bam file, 4055 reads had
pairs on different chromosomes

\hypertarget{what-is-the-insert-size-mean-and-standard-deviation}{%
\section{What is the insert size mean and standard
deviation?}\label{what-is-the-insert-size-mean-and-standard-deviation}}

According to the summary data of the NA20538.bam file, the mean insert
size was 190.3, and its standard deviation: 136.4

\end{document}
