---
title: "homework_05"
author: "Esteban Jorquera; Marlon Aldair; Diego Ram√≠rez"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_document: default
  css: readable.min.css
---

Github link:
https://github.com/EstebanJorquera/UNAM_BIOinfoII.git

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r results = TRUE, message=FALSE, warning=FALSE, echo=TRUE, Packages_Dependencies}
# Note installing packages through knitter is not the best idea,
# the following code will likely fail if the package is not already installed, with a missing mirror error message
# for this reason there's an included R script just for that
# anyway this meant to be run on a cluster rather than locally, so happy copy pasting :)!
# Installs required packages if not already installed (avoids re installing)
if (!requireNamespace("BiocManager", quietly = TRUE))     install.packages("BiocManager")

if (!requireNamespace("tidyverse", quietly = TRUE))       install.packages("tidyverse")
if (!requireNamespace("dplyr", quietly = TRUE))           install.packages("dplyr")
if (!requireNamespace("tidyr", quietly = TRUE))           install.packages("tidyr")
if (!requireNamespace("ggplot2", quietly = TRUE))         install.packages("ggplot2")
if (!requireNamespace("wordcloud", quietly = TRUE))       install.packages("wordcloud")
if (!requireNamespace("RColorBrewer", quietly = TRUE))    install.packages("RColorBrewer")

# Libraries
library(tidyverse)
library(dplyr)
library(tidyr)
library(ggplot2)
library(wordcloud)
library(RColorBrewer)

BiocManager::install(version = "3.13")
packages = c("DESeq2", "tximport")
BiocManager::install(packages)
```

```{code results = FALSE, message=FALSE, warning=FALSE, echo=TRUE, multi_qc.sge}
#!/bin/bash
# Use current working directory
#$ -cwd
#
# Join stdout and stderr
#$ -j y
#
# Run job through bash shell
#$ -S /bin/bash
#
# You can edit the script since this line
#
# Your job name
#$ -N EJ_multiqc

# Send an email after the job has finished
#$ -m e
#$ -M eijorquera@uc.cl
#
# Line required if modules are to be used, source modules environment
. /etc/profile.d/modules.sh
#
# quality control and trimming of reads,
# loads fastqc, trimmomatic and multiqc module, also downloads Truseq adapters
# fastqc is used to do qc analysis of the ChIP-seq reads
# multiqc generates a report for the non-trimmed reads
# wget downloads adapter file
# trimmomatic trimms the adapters from the reads using the adapter file as reference (ToDo check how to run a for loop in an sge script)
# fastqc is used to do qc analysis of the trimmed ChIP-seq reads 
# multiqc generates a report for the trimmed reads
(module load fastqc/0.11.3 ; 
module load trimmomatic/0.33 ; 
module load multiqc/1.5 ; 
fastqc /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/*.fastq.gz -o /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/QC1 ; 
multiQC /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/QC1 ; 
wget https://github.com/timflutre/trimmomatic/blob/master/adapters/TruSeq3-PE.fa ; 
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417885.sra_1.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417885_trmd_1.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417885.sra_2.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417885_trmd_2.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417886.sra_1.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417886_trmd_1.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417886.sra_2.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417886_trmd_2.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417887.sra_1.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417887_trmd_1.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417887.sra_2.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417887_trmd_2.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417888.sra_1.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417888_trmd_1.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417888.sra_2.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417888_trmd_2.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417889.sra_1.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417889_trmd_1.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417889.sra_2.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417889_trmd_2.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417890.sra_1.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417890_trmd_1.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
trimmomatic PE -phred33 -basein /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/data/fastq/SRR6417890.sra_2.fastq.gz -baseout /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/SRR6417890_trmd_2.fastq.gz ILLUMINACLIP:/mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:5:30 MINLEN:40 ;
fastqc /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/*.fastq.gz -o /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/QC2  
multiQC /mnt/Citosina/amedina/ejorquera/BioInfoII/Tarea_5/out/QC2)
```